### PR TITLE
Manuele benchmarking work (unfinished)

### PR DESCRIPTION
**Pull request by gridley**
Hi all,

I figured I'd go ahead and push my work on trying to reproduce Manuele's buoyant fast spectrum salt in a box case since it now comes with some code potentially helpful to others.

Here's what's included:

- data for running Manu's benchmark case (not output, although it does include an openFOAM mesh)
- eigenkernels for neutronics calculations that include delayed neutrons
- delayed neutrons now born in groups according to CHI_D rather than just the fast group
- A kernel for global heat removal proportional to a difference from some reference temperature, called ManuHX because he uses something like this in his benchmark case.
- PrecursorAction can now advect precursors according to Navier-Stokes coupled variables

The holdup on matching Manu's case comes from two things. Firstly, the neutronics must be run in eigenvalue mode with power normalized to a gigawatt. This is because we were only given cross sections at one temperature and thus can't rely on feedback to walk the salt box to steady-state. Secondly, 
trying to get reasonable reactivity in the neutrons only, [no precs case ](https://github.com/gridley/moltres/blob/manuBench/problems/publication_level_cases/manuBench/nts.i)using Manu's group constant seems to be an issue. I've checked over the new delayed neutron spectrum code several times and can't find any issues, so I think I need to plug Manu's group constants into another code to verify that either we're misinterpreting them in some way, or they actually result in a deeply subcritical eigenvalue.

### CODE REVIEW

**Comment by gridley** [comment]
I see that Civet refuses to run due to trailing whitespace; will be fixing this tomorrow.

**Comment by lindsayad** [comment]
So this is failing because all the tests are looking for property files corresponding to CHI_D that don't exist. There are a few options to fix this; maybe the best is to create a boolean parameter in `GenericMoltresMaterial` that is used to only look for `CHI_D` if `true`.

**Comment by gridley** [comment]
Yep, I was just getting to that today Alex. I'm thinking check for a CHI_D file, and if not present, assume delayed neutrons are born in the fastest group like the past. First off, I'm going to lump the piece of code that reads file data to make splines into one method, then have all the different spline types call that, since it seems repeated as-is.

**Comment by gridley** [comment]
they put the "close and comment" button way too close to the comment button :frowning:

**Comment by lindsayad** [comment]
@gridley I highly recommend you put a hook into your editor so that you don't have these trailing whitespace troubles.

File: src/materials/GenericMoltresMaterial.C [hunk 1]
```diff
@@ -162,13 +166,25 @@ GenericMoltresMaterial::dummyConstruct(std::string & property_tables_root,
...
     auto o = _vec_lengths[xsec_names[j]];
 
     _xsec_map[xsec_names[j]].resize(o);
     _xsec_spline_interpolators[xsec_names[j]].resize(o);
+
+    //chi_d backwards compatibility on unit tests:
+    if (xsec_names[j]=="CHI_D" and not myfile.good())
+    {
+      for (decltype(o) k = 0; k < o; ++k)
+        _xsec_map["CHI_D"][k] = (Real)0.0L;
```
Comment (lindsayad) [inline src/materials/GenericMoltresMaterial.C]: We don't want to specify a `long double` here. Additionally, there's no need to explicitly cast to Real (double). This is done implicitly. Just set equal to `0.` or `0.0`.

File: src/materials/GenericMoltresMaterial.C [hunk 2]
```diff
@@ -162,13 +166,25 @@ GenericMoltresMaterial::dummyConstruct(std::string & property_tables_root,
...
+
+      _xsec_map["CHI_D"][0] = (Real)1.0L;
```
Comment (lindsayad) [inline src/materials/GenericMoltresMaterial.C]: Same as above

**Comment by gridley** [comment]
Hey Alex,

I never thought to get this whitespace hook, great idea, thanks.

On the switch case, I had some trouble compiling the switch. It seems that MooseEnum equality comparison to strings works great, C++ switch cases require an integer interpretation. The troubles came when interpreting the MooseEnum as an int. After explicitly giving an enumeration (e.g. spline=1), _interp_type was taking values like -15000.

So, I figured I’d come back to that later and just get stuff working first.

File: include/materials/GenericMoltresMaterial.h [hunk 3]
```diff
@@ -40,6 +40,24 @@ class GenericMoltresMaterial : public GenericConstantMaterial
...
+    BICUBIC,
+    SPLINE,
+    MONOTONE_CUBIC,
+    LINEAR,
+    NONE,
+    LSQ
+  };
+
+  // returns a MooseEnum corresponding to the above enum
+  static MooseEnum interpTypes()
```
Comment (lindsayad) [inline include/materials/GenericMoltresMaterial.h]: 👍 for `static`

**Pull request merged**